x/programs/cmd/simulator/src/lib.rs

```diff
@@ -91,6 +93,27 @@ pub enum Key {
     Secp256r1(String),
 }
 
+#[derive(Clone, Debug, PartialEq, Default, Serialize)]
+#[serde(rename_all = "camelCase")]
+#[non_exhaustive]
+pub struct TestContext {
+    pub program_id: Id,
+}
+
+#[derive(Clone, Debug, PartialEq, Serialize)]
+#[serde(tag = "type", rename = "testContext")]
+pub(crate) struct SimulatorTestContext {
+    #[serde(serialize_with = "base64_struct_encode")]
+    value: TestContext,
+}
```

Comment:
It's clearly a hack. There is no way to externally tag a struct to have this `type` `value` structure like we have for enums.